src/wasm-binary.h

```diff
@@ -116,6 +116,7 @@ struct LEB {
   }
 };
 
+#define SIZE32LEB(v) U32LEB((uint32_t)(v))
```

Comment:
All `.size()` are size_t so a cast is needed for any 32-bit value. The reason for SIZE32LEB exists is to cast 64-bit size_t 32-bit LEB. I'm suspect that some of the int32_t in this file should be uint32_t or Index or some new name. I don't want to even peek at this issue until this PR is closed.